e3sm_diags/driver/tc_analysis_driver.py

```diff
@@ -163,13 +160,40 @@ def generate_tc_metrics_from_te_stitch_file(te_stitch_file: str) -> Dict[str, An
     """
     logger.info("\nGenerating TC Metrics from TE Stitch Files")
     logger.info("============================================")
+    if not os.path.exists(te_stitch_file):
+        raise FileNotFoundError(f"The file {te_stitch_file} does not exist.")
+
     with open(te_stitch_file) as f:
-        lines = f.readlines()
+        lines_orig = f.readlines()
+
+    if not lines_orig:
+        raise ValueError(f"The file {te_stitch_file} is empty.")
+
+    line_ind = []
+    data_start_year = int(te_stitch_file.split(".")[-2].split("_")[-2])
+    data_end_year = int(te_stitch_file.split(".")[-2].split("_")[-1])
+    for i in range(0, np.size(lines_orig)):
+        if lines_orig[i][0] == "s":
+            year = int(lines_orig[i].split("\t")[2])
+
+            if year <= data_end_year:
+                line_ind.append(i)
+
+    # Remove excessive time points cross year bounds from 6 hourly data
+    end_ind = line_ind[-1]
+    lines = lines_orig[0:end_ind]
```

Comment:
Minor suggestion, I would extract this block of logic into a private function called `_remove_time_crossing_bounds()` or something to make the main `run_diags()` cleaner. 

However, this can be done in the refactored codebase too.

Reply:
Thank you for the suggestion. I considered to have a function instead as well.  I think we can update the refactored code base when bringing more fix and enhancement to tc set.